### DISCUSSION TITLE
Accessing the cache entities

### DISCUSSION THREAD

**Discussion by arcanis** [urql-graphql/urql#1251] [Q&A] [answered]
Using `readFragment`, one can access an entity from its id. Is it similarly possible to access all entities from a given typename? I've tried `inspectField` with the given type as `__typename`, but it seems to only work for Query - anything else returns an empty array 🤔

My use case is that I have an operation that reorders a child entity inside its parent entity (each child entity has an `order` field, and my mutation mutates this field from all the entities it needs to) that I need to make optimistic. To do that, I'd like to iterate over all child entities in the cache to update their `order` field the same way the server would (by using a `writeFragment` for each affected entry). It however requires to have a way to know what's in the cache.

**Comment by kitten** [comment, marked as answer]
No, we currently don't store a list of all entities per type name and have looked into that before. However as of now we haven't quite gone that far yet but are still looking into it.

**Reply by arcanis** [reply]
Ok! For now I'm going the ugly way by directly accessing `cache.data.records.base` w/ a `startWith` check to get their IDs, and using `readFragment` on each of them to populate the fields I want.
